src/main/java/com/example/airbnbapi/service/MediaService.java

```diff
@@ -91,6 +94,109 @@ public boolean itemExists(MediaType type, String title) {
         }
         return false;
     }
+
+    //// authentication ////
+
+    // add an authenticated user to the authenticated db
+    public OauthUser authenticateUser(User user) {
+
+        if (authenticatedUserExists(user.getEmailAddress())) {
+            return null;
+        }
+
+
+        try {
+
+            String hashedPassword = BCrypt.hashpw(user.getPassword(), BCrypt.gensalt());
+
+            OauthUser oauthUser = new OauthUser();
+            oauthUser.setOauthCode(UUID.randomUUID());
+            oauthUser.setEmailAddress(user.getEmailAddress());
+            oauthUser.setPassword(hashedPassword);
+
+            return oauthUserRepository.save(oauthUser);
+        } catch (RuntimeException e) {
+
+            logger.error("An error occurred when trying to request data from the database, RuntimeException: " + e);
+            throw new DataBaseException("Sorry, couldn't establish connection to the database", e);
+        }
+
+    }
+
+    // adds a user to the user db.
+    public User addUser(User user) {
+
+        List<User> allUsers = userRepository.findAll();
```

Comment:
Why do you need to findAll users to add a new user? Perhaps you could query the db to check if the email already exists and if it doesn't, save?